lib/jekyll/page.rb

```diff
@@ -174,8 +162,39 @@ def trigger_hooks(hook_name, *args)
       Jekyll::Hooks.trigger :pages, hook_name, self, *args
     end
 
-    def write?
+    # Don't generate excerpts for pages
+    def generate_excerpt?
+      false
+    end
+
+    # Determine whether the file should be rendered with Liquid.
+    #
+    # Always returns true.
+    def render_with_liquid?
       true
     end
+
+    # Don't generate a title
+    def post_read
+      original = data['title']
+      super
+      data['title'] = original
+    end
+
+    # NOTE: COMPATIBILITY
+    #
+    # Accessor for data properties by Liquid.
+    #
+    # property - The String name of the property to retrieve.
+    #
+    # Returns the String value or nil if the property isn't included.
+    def [](property)
+      if ATTRIBUTES_FOR_LIQUID.include?(property)
+        send(property)
```

Comment:
Even though it matters little since you are already guarding, the double guard (`public_send`) would be great!